app/models/manager_refresh/inventory_collection_default/infra_manager.rb

```diff
@@ -409,18 +409,26 @@ def lans(extra_attributes = {})
     def snapshot_parent(extra_attributes = {})
       snapshot_parent_save_block = lambda do |_ems, inventory_collection|
         snapshot_collection = inventory_collection.dependency_attributes[:snapshots].try(:first)
+        snapshot_parents    = snapshot_collection.data.each_with_object({}) do |snapshot, hash|
+          hash[snapshot.id] = snapshot.parent_uid if snapshot.parent_uid
+        end
+
+        ActiveRecord::Base.transaction do
+          snap_recs   = Snapshot.find(snapshot_parents.keys)
```

Comment:
@agrare @borod108 right, so we are seeing the faihlure, because snapshots are deleted as a tree

So thew spec fails, because we delete the parent snapshot. So if I manually delete the parent snapshot by ```snapshot.destroy``` I see that the all snapshots below are deleted, leading to ```Snapshot.count => 0```

I think we need to do ```snap_recs = Snapshot.where(snapshot_parents.keys).all```

Reply:
specs in https://github.com/ManageIQ/manageiq-providers-ovirt/pull/241/files